src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -93,12 +93,12 @@ public async Task StartAsync(CancellationToken cancellationToken)
                 }
             }
 
-            await _mediator.Publish(new SearchParametersUpdated(updated));
+            await _mediator.Publish(new SearchParametersUpdated(updated), cancellationToken);
         }
 
-        public Task StopAsync(CancellationToken cancellationToken)
+        public async Task Handle(StorageInitializedNotification notification, CancellationToken cancellationToken)
         {
-            return Task.CompletedTask;
+            await StartAsync(cancellationToken);
```

Comment:
nit: Perhaps we should return the name of StartAsync to EnsureInitialized, as it was before? Since there is not a Start/Stop .

Reply:
It could probably be internal too since it is only invoked from tests now